client/app/scripts/superdesk-search/search.js

```diff
@@ -421,6 +421,23 @@
                         }
                     }
                 });
+                /*
+                * if one of the parameters is the original creator then replace the user id with the
+                * display name.
+                */
+                var i = -1;
+                for (i in tags.selectedParameters) {
```

Comment:
Using the `for..in` construct to loop over an array might result in unexpected behavior, thus I suggest to rewrite this to e.g. use the Array's `forEach()` method or some other looping construct.

**Explanation:** Let's say some module adds a new enumerable property to the Array's prototype, e.g. a convenient utility method. The `for..in` construct would include such property, too!

```js
>>> Array.prototype.foo = 'bar';
>>> var a = [1,2,3];
>>> for (var i in a) {
...     console.log(i);
... }
0
1
2
foo  # <--- oops!
```